code/modules/client/preference_setup/loadout/loadout.dm

```diff
@@ -60,15 +60,21 @@ var/list/gear_datums = list()
 	var/mob/preference_mob = preference_mob()
 	for(var/gear_name in gear_datums)
 		var/datum/gear/G = gear_datums[gear_name]
-		var/okay = 1
+		var/okay = TRUE
 		if(G.whitelisted && preference_mob)
-			okay = 0
+			okay = FALSE
 			// TODO: enable after baymed
 			/*for(var/species in G.whitelisted)
 				if(is_species_whitelisted(preference_mob, species))
 					okay = 1
 					break
 					*/
+		// This has to return TRUE if there is no preference_mob or client
+		// because they don't exist when we're sanitizing the gear during pref load
+		// so it'll make it impossible to save the slot
+		if(preference_mob && preference_mob.client)
+			if(G.ckey_whitelist && !(preference_mob.ckey in G.ckey_whitelist))
```

Comment:
It may be worth considering changing the first part of that conditional to `islist(G.ckey_whitelist)` on the off chance it's set to something dumb.